core/src/main/java/net/automatalib/modelchecking/modelchecker/cache/InternalModelCheckerDelegator.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright (C) 2013-2018 TU Dortmund
+ * This file is part of AutomataLib, http://www.automatalib.net/.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.automatalib.modelchecking.modelchecker.cache;
+
+import java.util.Collection;
+
+import net.automatalib.automata.transout.MealyMachine;
+import net.automatalib.modelchecking.Lasso;
+import net.automatalib.modelchecking.Lasso.MealyLasso;
+import net.automatalib.modelchecking.ModelChecker;
+import net.automatalib.modelchecking.ModelChecker.MealyModelChecker;
+import net.automatalib.modelchecking.ModelCheckerLasso;
+import net.automatalib.modelchecking.ModelCheckerLasso.MealyModelCheckerLasso;
+
+/**
+ * Internal utility interface to provide default implementations for {@link ModelChecker}s based on a given delegator
+ * (cf. {@link #getModelChecker()}). Being based in interfaces, this interface and its inner sub-interfaces allow for a
+ * kind of multiple inheritance.
+ *
+ * @param <MC>
+ *         model checker type
+ * @param <I>
+ *         input alphabet type
+ * @param <A>
+ *         automaton type
+ * @param <P>
+ *         property type
+ * @param <R>
+ *         counterexample type
+ *
+ * @author frohme
+ */
+interface InternalModelCheckerDelegator<MC extends ModelChecker<I, A, P, R>, I, A, P, R> {
```

Comment:
Shouldn't InternalModelCheckerDelegator be a ModelChecker and provide a default implementation for findCounterExample? ModelCheckerLassoDelegator is a ModelCheckerLasso too.

Reply:
If this was a public interface (note: it is package private) that others should use, I'd agree with you. But currently this is only used to propagate the necessity for a `getModelChecker` method in the sub-interfaces. It is only used in the context of the `SizedModelCheckerCache` which implements the `findCounterExample` method anyway. So adding this method here, would just add a redundant default method that gets overridden anyway. If in the future there is a proper use-case for this, I'm happy to add this.